### PR TITLE
mockkubeapiserver: support PartialObjectMetadata watches

### PR DESCRIPTION
**Pull request by justinsb**
controller-runtime seems to have some bugs around apiservers that
don't support this, so introduce a minimal implementation.

### CODE REVIEW

**Comment by justinsb** [comment]
/assign @apelisse 

Would be great to do something better here, but this seems like the minimal implementation and otherwise controller-runtime is unhappy

File: mockkubeapiserver/listresource.go [hunk 1]
```diff
@@ -46,9 +49,16 @@ func (req *listResource) Run(ctx context.Context, s *MockKubeAPIServer) error {
 
 	query := req.r.URL.Query()
 
+	// TODO: we should parse the accept header correctly
+	partialObjectMetadata := false
+	accept := req.r.Header.Get("accept")
+	if strings.Contains(accept, "PartialObjectMetadata") {
+		partialObjectMetadata = true
+	}
```
Comment (apelisse) [inline mockkubeapiserver/listresource.go:57]: Nit:
```suggestion
	accept := req.r.Header.Get("accept")
	partialObjectMetadata := strings.Contains(accept, "PartialObjectMetadata")
```

**Review by apelisse — COMMENTED** [review]
/lgtm

**Review by apelisse — COMMENTED** [review]
/lgtm

**Pull request merged**